mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -0,0 +1,793 @@
+
+			if filter.TermsNotInBounds != "" {
+				mapCharFilters[i]["terms_not_in_bounds"] = filter.TermsNotInBounds
+			}
+
+			if filter.MinShingleSize != nil {
+				mapCharFilters[i]["min_shingle_size"] = *filter.MinShingleSize
+			}
+
+			if filter.MaxShingleSize != nil {
+				mapCharFilters[i]["max_shingle_size"] = *filter.MaxShingleSize
+			}
+
+			if filter.Pattern != "" {
+				mapCharFilters[i]["pattern"] = filter.Pattern
+			}
+
+			if filter.Replacement != "" {
+				mapCharFilters[i]["replacement"] = filter.Replacement
+			}
+
+			if filter.Matches != "" {
+				mapCharFilters[i]["matches"] = filter.Matches
+			}
+
+			if filter.StemmerName != "" {
+				mapCharFilters[i]["stemmer_name"] = filter.StemmerName
+			}
+
+			if len(filter.Tokens) > 0 {
+				mapCharFilters[i]["tokens"] = filter.Tokens
+			}
+
+			if filter.IgnoreCase != nil {
+				mapCharFilters[i]["ignore_case"] = *filter.IgnoreCase
+			}
+
+		}
+		return mapCharFilters
+	}
+	return nil
+}
+
+func flattenSearchIndexCharFilters(filters []*matlas.AnalyzerCharFilter) []map[string]interface{} {
+
+	if len(filters) > 0 {
+		mapCharFilters := make([]map[string]interface{}, len(filters))
+
+		for i, filter := range filters {
+			mapCharFilters[i] = map[string]interface{}{
+				"type": filter.Type,
+			}
+
+			if len(filter.IgnoreTags) > 0 {
+				mapCharFilters[i]["ignore_tags"] = filter.IgnoreTags
+			}
+
+			if filter.Mappings != nil {
+				mapCharFilters[i]["mappings"] = marshallSearchIndexCharFilterMappingFields(*filter.Mappings)
+			}
+
+		}
+		return mapCharFilters
+	}
+	return nil
+}
+
+func resourceMongoDBAtlasSearchIndexCreate(d *schema.ResourceData, meta interface{}) error {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+	projectID := d.Get("project_id").(string)
+
+	clusterName := d.Get("cluster_name").(string)
+
+	searchIndexRequest := &matlas.SearchIndex{
+		Analyzer:       d.Get("analyzer").(string),
+		Analyzers:      expandCustomAnalyzers(d.Get("analyzers").([]interface{})), //TODO: check if is correct type
+		CollectionName: d.Get("collection_name").(string),
+		Database:       d.Get("database").(string),
+		Mappings: &matlas.IndexMapping{
+			Dynamic: d.Get("mappings_dynamic").(bool),
+			Fields:  unmarshalSearchIndexMappingFields(d.Get("mappings_fields").(string)),
+		},
+		Name:           d.Get("name").(string),
+		SearchAnalyzer: d.Get("search_analyzer").(string),
+		Status:         d.Get("status").(string),
+	}
+
+	dbSearchIndexRes, _, err := conn.Search.CreateIndex(context.Background(), projectID, clusterName, searchIndexRequest)
+	if err != nil {
+		return fmt.Errorf("error creating database user: %s", err)
+	}
+
+	d.SetId(encodeStateID(map[string]string{
+		"project_id":   projectID,
+		"cluster_name": clusterName,
+		"index_id":     dbSearchIndexRes.IndexID,
+	}))
+
+	return resourceMongoDBAtlasSearchIndexRead(d, meta)
+}
+
+func customAnalyzersSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet, //TODO: change type
```

Comment:
Are the TODOs done?  If not they should be before review, if they are the comment should be removed.

Reply:
I forgot to remove it